dali/pipeline/data/tensor_list.h

```diff
@@ -755,7 +755,7 @@ class DLL_PUBLIC TensorList {
   }
 
  protected:
-  Buffer<Backend> data_;
+  Buffer<Backend> data_ = {};
```

Comment:
```suggestion
  Buffer<Backend> data_{};
```
or leave it as-is - it's a false positive.

Reply:
I believe, that in this case both
```c++
Buffer<Backend data_ = {};
Buffer<Backend> data_{};
```
are the same:

`Buffer<Backend> data_{}` performs [value initialization](https://en.cppreference.com/w/cpp/language/value_initialization) per syntax (5). `Buffer<Backend> data_ = {};` is a [copy-list-initialization](https://en.cppreference.com/w/cpp/language/list_initialization) per syntax (10) and since `Buffer<>` is not an aggregate type, the value initialization is performed (see `Explanation` in https://en.cppreference.com/w/cpp/language/list_initialization):
> Otherwise, if the braced-init-list is empty and T is a class type with a default constructor, [value-initialization](https://en.cppreference.com/w/cpp/language/value_initialization) is performed.

Reply:
Well.... theoretically. I specifically remember seeing bugs related to overspecifing initialization of an object having a default constructor (which resulted in an attempt to copy). I don't remember which compiler did this (clang, gcc or nvcc).